synapseclient/__main__.py

```diff
@@ -289,6 +291,18 @@ def onweb(args, syn):
 
 def setProvenance(args, syn):
     """Set provenance information on a synapse entity."""
+    if isinstance(args.id, str) and os.path.isfile(args.id):
+        results = syn.restGET('/entity/md5/%s' % utils.md5_for_file(args.id).hexdigest())['results']
```

Comment:
Pull out lines 295-302 into a helper function and re-use 4x times across set/get provenance and annotations?